azure-pipelines/azure-pipelines.yml

```diff
@@ -645,3 +645,140 @@ stages:
           --build-arg PROJECT_ROOT="$(System.DefaultWorkingDirectory)" $(System.DefaultWorkingDirectory) \
           -t edgehub/mockdevice-opcua:$(release_tag) --push
         displayName: build edgehub/mockdevice-opcua
+
+- stage: docker_build_shifu_demo_push
+  condition: and(succeeded(), variables['build.retainedByRelease'])
```

Comment:
what does `build.retainedByRelease` do in this case?

Reply:
this stage only run when shifu release ..  the others will skip